apps/src/lib/util/UserPreferences.js

```diff
@@ -163,31 +163,86 @@ export default class UserPreferences extends Record({userId: 'me'}) {
     }
   }
 
-  async getGlobalTheme(errorCallback) {
+  /**
+   * Fetches all user theme settings (e.g., global, blockly).
+   * @param {function} [errorCallback]
+   */
+  async getThemeSettings(errorCallback) {
     try {
       const themeResponse = await HttpClient.fetchJson(
         '/user_preference/theme'
       );
-      return themeResponse.value?.theme?.global;
+      return themeResponse.value?.theme;
     } catch (error) {
       // Don't call the error callback if 'Not found', as it just means the
       // user has not set a theme yet.
-      if (error.response.status !== 404) {
-        errorCallback(error);
+      if (error?.response?.status !== 404) {
+        return errorCallback(error) ?? null;
       }
       return null;
     }
   }
 
-  async setGlobalTheme(theme) {
-    const body = {
-      theme: {
-        global: theme,
-      },
-    };
+  /**
+   * Fetches the user's global theme preference.
+   * @param {function} [errorCallback]
+   */
+  async getGlobalTheme(errorCallback) {
+    const theme = await this.getThemeSettings(errorCallback);
+    return theme?.global ?? null;
+  }
 
-    return HttpClient.put('/user_preference', JSON.stringify(body), true, {
-      'Content-Type': 'application/json',
-    });
+  /**
+   * Retrieves the user's Blockly theme preference.
+   * @param {function} [errorCallback]
+   */
+  async getBlocklyTheme(errorCallback) {
+    const theme = await this.getThemeSettings(errorCallback);
+    return theme?.blockly ?? null;
+  }
+
+  /**
+   * Merges new theme settings with any existing preferences and saves them.
+   *
+   * @param {Object} themeUpdate - A partial theme object to merge (e.g., {global: 'Dark'}).
+   * @param {function} [errorCallback]
+   */
+  async updateThemeSettings(themeUpdate, errorCallback) {
+    try {
+      const existingTheme = await this.getThemeSettings(() => {});
+      const updatedTheme = {
```

Comment:
I don't think we need to fetch the existing theme. The [controller](https://github.com/code-dot-org/code-dot-org/blob/11b4a4463711143529ee8de95df79a52df54d0c2/dashboard/app/controllers/user_preferences_controller.rb#L16-L18) merges in the requested theme with the existing theme.